marimo/_server/api/endpoints/datasources.py

```diff
@@ -104,3 +105,29 @@ async def preview_sql_table_list(request: Request) -> BaseResponse:
         from_consumer_id=ConsumerId(app_state.require_current_session_id()),
     )
     return SuccessResponse()
+
+
+@router.post("/fetch_datasource_connection")
+@requires("edit")
+async def fetch_datasource_connection(request: Request) -> BaseResponse:
```

Comment:
Should this be renamed to PreviewX similar to the others? Or is there something different.